### PR TITLE
Add Mockiato to the shootout

### PR DESCRIPTION
**Pull request by asomers**
CC @bash .  This will add mockiato to the mock shootout.  Could you please look over lib/t_mockiato.rs and tell me if there any features I missed?

### CODE REVIEW

File: src/t_mockiato.rs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+        mock.expect_foo().returns(2i16);
+        assert_eq!(mock.foo(), 2);
+    }
+
+    // Mockiato implemented this feature only for one type: ()
+    fn return_default() {
+        unimplemented!()
+    }
+
+    fn return_panic() {
```
Comment (tautropfli) [inline src/t_mockiato.rs:303]: This is [supported](https://docs.rs/mockiato/0.9.0/mockiato/struct.MethodCallBuilder.html#method.panics):
```rust
let mut mock = AMock::new();
mock.expect_foo()
    .panics_with_message("Some panic message")
    .times(..);
```

File: src/t_mockiato.rs [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+        pub trait A {
+            fn into_nothing(self);
+        }
+
+        let mut mock = AMock::new();
+        mock.expect_into_nothing();
+        mock.into_nothing();
+    }
+
+    fn external_trait() {
```
Comment (tautropfli) [inline src/t_mockiato.rs:69]: This is possible using the [`remote`](https://docs.rs/mockiato/0.9.0/mockiato/macro.mockable.html#remote) parameter on the `#[mockable]` attribute:

```rust
#[cfg(test)]
#[mockable(remote = "io::Write")]
trait Write {
    fn write(&mut self, buf: &[u8]) -> io::Result<usize>;

    fn flush(&mut self) -> io::Result<()>;

    // Methods with a default implementation can be omitted.
}
```

File: src/t_mockiato.rs [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        // traits.
+        unimplemented!()
+    }
+
+    fn return_call_with_args() {
+        unimplemented!()
+    }
+
+    fn return_reference() {
+        // Mockiato can only return 'static references
```
Comment (tautropfli) [inline src/t_mockiato.rs]: This is only true when `#[mockable(static_references)]` is enabled.
Example of returning a non-static reference:
https://github.com/myelin-ai/mockiato/blob/master/tests/return_values_with_references_work.rs

File: src/t_mockiato.rs [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+    fn version() {
+        let ver = crate::built_info::DEPENDENCIES.iter()
+            .find(|(name, _)| *name == "mockiato")
+            .unwrap()
+            .1;
+        print!("{} ", ver);
+    }
+
+    fn where_clause() {
+        // I think where clauses work but only for generic traits, not generic
```
Comment (tautropfli) [inline src/t_mockiato.rs:401]: Yes, where clauses work for generic traits:
https://github.com/myelin-ai/mockiato/blob/a1a06c8fea51f10bc22182b06a691b4ea2c91dfb/tests/generic_types.rs#L10

**Review by tautropfli — COMMENTED** [review]
Thank you for adding mockiato to this list 😀

File: src/t_mockiato.rs (hunk 1 above)
Comment (asomers) [reply src/t_mockiato.rs:303]: Ok.  I didn't notice that in the docs, but I see it now.

File: src/t_mockiato.rs (hunk 2 above)
Comment (asomers) [reply src/t_mockiato.rs:69]: Go it.

File: src/t_mockiato.rs (hunk 3 above)
Comment (asomers) [reply src/t_mockiato.rs]: I see now.

**Comment by asomers** [comment]
Thanks for the help.  Does Mockiato have any unique features that don't exist in the shootout?

**Pull request merged**